src/lib/y2storage/planned/partitions_distribution.rb

```diff
@@ -111,31 +117,77 @@ def partitions_count
       #
       # @return [Integer] -1, 0, 1 like <=>
       def better_than(other)
-        # The smallest gaps the better
-        res = gaps_total_size <=> other.gaps_total_size
-        return res unless res.zero?
+        criteria = [
+          # The smaller gaps the better
+          [:gaps_total_size, :smaller],
 
-        # The fewer gaps the better
-        res = gaps_count <=> other.gaps_count
-        return res unless res.zero?
+          # The fewer gaps the better
+          [:gaps_count, :smaller],
 
-        # The fewer physical volumes the better
-        res = partitions_count <=> other.partitions_count
-        return res unless res.zero?
+          # The fewer physical volumes the better
+          [:partitions_count, :smaller],
 
-        # The less fragmentation the better
-        res = spaces.count <=> other.spaces.count
-        return res unless res.zero?
+          # The bigger installation the better
+          [:spaces_total_size, :bigger],
 
-        # The biggest installation the better
-        res = other.spaces_total_size <=> spaces_total_size
-        return res unless res.zero?
+          # We want the partitions with biggest weight to be placed in the spaces
+          # in which there is more extra space to be distributed
+          [:weight_space_deviation, :smaller],
+
+          # The less fragmentation the better
+          [:spaces_count, :smaller]
+        ]
```

Comment:
btw this start smelling for `Comparator` class, which gets two object and compare it, so you can have even more complex criteria. and does not need to have that helpers.